pkg/cloud/handlers.go

```diff
@@ -60,6 +61,32 @@ func RecordRequestsMiddleware() func(*middleware.Stack) error {
 	}
 }
 
+// LogServerErrorsMiddleware is a middleware that logs server errors received when attempting to contact the AWS API
+// A specialized middleware is used instead of the SDK's built-in retry logging to allow for customizing the verbosity
+// of throttle errors vs server/unknown errors, to prevent flooding the logs with throttle errors
+func LogServerErrorsMiddleware() func(*middleware.Stack) error {
```

Comment:
Instead of creating a new middleware function, could we use the existing [RecordRequestsMiddleware](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/e8df5a884b866cebd4e09a85354fd851189cde63/pkg/cloud/handlers.go#L51) (which already logs throttled errors) to handle the logging?

For example, non-throttled errors could be logged within this conditional: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/e8df5a884b866cebd4e09a85354fd851189cde63/pkg/cloud/handlers.go#L50-L52

Reply:
Personally, I think it's better to leave these separate as even if it costs us a few lines of code, it separates the concerns and makes the logic much easier to understand compared to mixing the two.

That said, it's obviously possible to combine them so I'll leave this open to gather consensus on the best approach.

Reply:
Should we decide to proceed with this change then would it make sense to at least migrate that log statement from `RecordRequestsMiddleware` to this new function?

Reply:
Yes - 100%. Whichever decision, there should only be one place for logging the errors.

Reply:
+1 on separating concerns, and making RecordRequests not have the logging side effect.

Reply:
It seems odd duplicating an entire middleware function for a log statement but I support the team's decision, resolving.